src/kaitai_spec/attribute/contents.rs

```diff
@@ -0,0 +1,81 @@
+use serde::{Deserialize, Deserializer};
+use serde::de::{self, MapAccess, Visitor};
+use std::fmt;
+
+
+#[derive(Debug, Deserialize)]
+#[serde(untagged)]
+pub enum StringOrByte<'a> {
+    String(&'a str),
+    Byte(u8),
+}
+
+#[derive(Debug)]
+pub struct ContentsBytes(pub Vec<u8>);
+
+impl<'de> Deserialize<'de> for ContentsBytes {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct MyDataVisitor;
+
+        impl<'de> de::Visitor<'de> for MyDataVisitor {
+            type Value = ContentsBytes;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("a sequence of bytes or strings")
+            }
+
+            fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+            where
+                E: de::Error,
+            {
+                Ok(ContentsBytes(value.as_bytes().to_vec()))
+            }
+
+            fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+            where
+                A: de::SeqAccess<'de>,
+            {
+                let mut bytes = Vec::with_capacity(1_000);
+                while let Some(string_or_byte) = seq.next_element::<StringOrByte>()? {
+                    match string_or_byte {
+                        StringOrByte::Byte(b) => bytes.push(b),
+                        StringOrByte::String(string) => {
+                            bytes.extend_from_slice(string.as_bytes());
+                        }
+                    }
+                }
+                Ok(ContentsBytes(bytes))
```

Comment:
from my point of view it is better to separate return statement with an empty line. For better readability